### PR TITLE
Feature - Add FS project

### PR DESCRIPTION
**Pull request by at-thinhung**

### CODE REVIEW

**Comment by at-linhvo** [comment]
@at-thinhung please make core api with rx 🥇  Thank you 👍

File: FS/FS/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        let homeController = HomeViewController()
+        let homeNavigation = UINavigationController(rootViewController: homeController)
+        let searchController = SearchViewController()
+        let searchNavigation = UINavigationController(rootViewController: searchController)
+        let favoriteController = FavoriteViewController()
+        let favoriteNavigation = UINavigationController(rootViewController: favoriteController)
+        let profileController = ProfileViewController()
+        let profileNavigation = UINavigationController(rootViewController: profileController)
+
+        let tabbarController = UITabBarController()
```
Comment (at-suho) [inline FS/FS/AppDelegate.swift]: I think we need a class like `TabBarController` extend UITabBarController,  and move all config of tabbar into it,  AppDelegate just have: 
```swift
let tabbarController  = TabBarController()
window?.rootViewController = tabbarController
window?.makeKeyAndVisible()
```
So we move code from line 20 to 27 into class `TabBarController`

File: FS/FS/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        tabbarController.viewControllers = [homeNavigation, searchNavigation, favoriteNavigation, profileNavigation]
+        window?.rootViewController = tabbarController
+        window?.makeKeyAndVisible()
+        return true
+    }
+
+    func applicationWillResignActive(_ application: UIApplication) {
```
Comment (at-sunguyen) [inline FS/FS/AppDelegate.swift]: Please remove un-used code.

File: FS/FS/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
+        // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
+    }
+
+    func applicationDidEnterBackground(_ application: UIApplication) {
```
Comment (at-sunguyen) [inline FS/FS/AppDelegate.swift]: same above.

File: FS/FS/AppDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
+        // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+    }
+
+    func applicationWillEnterForeground(_ application: UIApplication) {
```
Comment (at-sunguyen) [inline FS/FS/AppDelegate.swift]: same above.

File: FS/FS/AppDelegate.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
+    }
+
+    func applicationDidBecomeActive(_ application: UIApplication) {
```
Comment (at-sunguyen) [inline FS/FS/AppDelegate.swift]: same above.

File: FS/FS/AppDelegate.swift [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+    }
+
+    func applicationWillTerminate(_ application: UIApplication) {
```
Comment (at-sunguyen) [inline FS/FS/AppDelegate.swift]: same above.

File: FS/FS/Classes/Model/Venue.swift [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import ObjectMapper
+
+class Venue: Mappable {
+    required init?(map: Map) {
+        
+    }
+
+    var id: String?
+    var name = "name"
+    func mapping(map: Map) {
```
Comment (at-sunguyen) [inline FS/FS/Classes/Model/Venue.swift]: Need a break line between properties and functions

File: FS/FS/Classes/View/Controller/Home/HomeViewController.swift [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+class HomeViewController: ViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Home"
+        API.getVenues(params: [:]).subscribe { (event) in
+            print(event)
+        }
+        .disposed(by: self.disposeBag)
```
Comment (at-quangphu) [inline FS/FS/Classes/View/Controller/Home/HomeViewController.swift]: `self` no need here. You can remove or not. You're boss. 🔪

File: FS/FS/AppDelegate.swift (hunk 1 above)
Comment (at-thinhung) [reply FS/FS/AppDelegate.swift]: @at-suho 
Done!

File: FS/FS/AppDelegate.swift (hunk 2 above)
Comment (at-thinhung) [reply FS/FS/AppDelegate.swift]: @at-sunguyen 
Done!

File: FS/FS/Classes/Model/Venue.swift (hunk 7 above)
Comment (at-thinhung) [reply FS/FS/Classes/Model/Venue.swift]: @at-sunguyen 
Done!

File: FS/FS/Classes/View/Controller/Home/HomeViewController.swift (hunk 8 above)
Comment (at-thinhung) [reply FS/FS/Classes/View/Controller/Home/HomeViewController.swift]: @at-quangphu 
Done!

**Pull request merged**